wallet/src/de/schildbach/wallet/ui/more/tools/ZenLedgerViewModel.kt

```diff
@@ -0,0 +1,105 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.ui.more.tools
+
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import de.schildbach.wallet.Constants
+import org.bitcoinj.core.Address
+import org.bitcoinj.script.ScriptPattern
+import org.bitcoinj.wallet.KeyChain
+import org.dash.wallet.common.WalletDataProvider
+import org.dash.wallet.common.services.BlockchainStateProvider
+import org.dash.wallet.common.util.Constants.DASH_CURRENCY
+import org.slf4j.LoggerFactory
+import javax.inject.Inject
+
+@HiltViewModel
+class ZenLedgerViewModel @Inject constructor(
+    private val walletDataProvider: WalletDataProvider,
+    private val blockchainStateProvider: BlockchainStateProvider,
+    private val zenLedgerClient: ZenLedgerClient
+) : ViewModel() {
+
+    companion object {
+        private val log = LoggerFactory.getLogger(ZenLedgerViewModel::class.java)
+    }
+    suspend fun isSynced(): Boolean {
+        return blockchainStateProvider.getState()?.isSynced() ?: false
+    }
+
+    var signUpUrl: String? = null
+
+    suspend fun sendTransactionInformation(): Boolean {
+        val wallet = walletDataProvider.wallet!!
+        val transactions = wallet.getTransactions(false)
+        val addresses = if (transactions.isEmpty()) {
+            listOf(
+                ZenLedgerAddress(
+                    DASH_CURRENCY,
+                    DASH_CURRENCY,
+                    wallet.currentAddress(KeyChain.KeyPurpose.RECEIVE_FUNDS).toBase58(),
+                    "Dash Wallet"
+                )
+            )
+        } else {
+            val addresses = arrayListOf<ZenLedgerAddress>()
+            transactions.forEach { tx ->
+                tx.outputs.forEach { output ->
+                    if (output.isMine(wallet)) {
+                        log.info(
+                            output.value.toFriendlyString(),
+                            Address.fromPubKeyHash(
+                                Constants.NETWORK_PARAMETERS,
+                                ScriptPattern.extractHashFromP2PKH(output.scriptPubKey)
+                            ).toBase58()
+                        )
+                        addresses.add(
+                            ZenLedgerAddress(
+                                DASH_CURRENCY,
+                                DASH_CURRENCY,
+                                Address.fromPubKeyHash(
+                                    Constants.NETWORK_PARAMETERS,
+                                    ScriptPattern.extractHashFromP2PKH(output.scriptPubKey)
+                                ).toBase58(),
+                                "Dash Wallet"
+                            )
+                        )
+                    }
+                }
+            }
+            addresses
+        }
```

Comment:
If there are no transactions, then we will send the current receive address.  This way the flow can proceed to the end with ZenLedger.  If no addresses are submitted, then ZenLedger returns an error with their create portfolio API.

If there are transactions, then we will scan all transaction outputs and add the addresses of the outputs that the wallet owns.